### PR TITLE
Fix centering issue with hacky spacer

### LINKED ISSUE #9870: Number visualization title on dashboard card is off-center when tooltip exists
For number visualizations on dashboards, the tooltip "i" icon is included in the centering calculation for the title. Before a user hovers and sees the icon appear, the title appears off-center skewed toward the left.

I see the same behavior on Firefox 66 and Chrome 73 (both on MacOS 10.14) running Metabase 0.32.4

Screenshots attached.

# No Hover
no-hover

# Hover
hover

### PR DESCRIPTION
**Pull request by paulrosenzweig**
Resolves #9870 

# Before

before

# After

after

# Any better options?

I think this looks better, but I don't really like the solution. It's simple and only a line of code, but adding a spacer that needs to match the width of the tooltip icon feels awkward to me.

I also considered absolutely positioning the icon, but that required a bunch more pixel tweaking than this. Are there any clever flexbox based solutions that I'm missing?

### CODE REVIEW

**Comment by tlrobinson** [comment]
The best I could come up with is 3 flex element in the row, with the side items set to `flex: 1 0 0`, but I'm not sure that's much better: https://codepen.io/tlrobinson/pen/XLWmeL

@kdoh?

**Comment by kdoh** [comment]
Only thing I might do is set a const for the "magic" width number and apply it to both that spacer and the icon so they stay in sync (theoretically someone could bust this centering by changing the default size prop on the `Icon` component) and there's one more marker of how the two are tied together going on.

**Comment by paulrosenzweig** [comment]
@kdoh I considered putting the same icon on the left too but hiding it. That would nicely keep thing in sync, but felt even hackier.

I like your idea of forcing the widths to match. It'll be a bit weird because the full 24px is made up of the icon width and a standard margin class (`.ml1`). I should be able to strip that out and keep things looking good though 👍.

File: frontend/src/metabase/visualizations/components/ScalarValue.jsx [hunk 1]
```diff
@@ -27,30 +27,40 @@ const ScalarValue = ({ value, isFullscreen, isDashboard }) => (
...
-  <div className="flex align-center full justify-center px2">
-    <h3
-      onClick={onClick}
-      className={cx(
-        "Scalar-title overflow-hidden fullscreen-normal-text fullscreen-night-text text-brand-hover",
-        {
-          "cursor-pointer": !!onClick,
-        },
+export const ScalarTitle = ({ title, description, onClick }) => {
+  const iconWidth = "24px";
```
Comment (tlrobinson) [inline frontend/src/metabase/visualizations/components/ScalarValue.jsx]: Typically I'd make this a top-level constant, like `const ICON_WIDTH = 24;`. Also you can leave off `px` when providing px values in React `style` props.

**Review by tlrobinson — APPROVED** [review]

**Pull request merged**